### ISSUE TITLE
_find_linker_path returns wrong path

### ISSUE THREAD

**Issue by zakcutner** [bazelbuild/bazel#16053]
### Description of the bug:

The [`_find_linker_path` function](https://github.com/bazelbuild/bazel/blob/master/tools/cpp/unix_cc_configure.bzl#L175-L219) determines the linker path by using the output of the following command (e.g. for the `lld` linker):
```bash
$ cc empty.cc -o /dev/null -Wl,--start-lib -Wl,--end-lib -fuse-ld=lld -v
```
It then looks for a line with `lld` in (since this is the linker we are looking for). The problem is that on my system, this also matches `-debugger-tuning=lldb`:
```bash
$ cc empty.cc -o /dev/null -Wl,--start-lib -Wl,--end-lib -fuse-ld=lld -v 2>&1 | rg lld
 "/usr/local/Cellar/llvm/14.0.6_1/bin/clang-14" -cc1 -triple x86_64-apple-macosx12.0.0 -Wundef-prefix=TARGET_OS_ -Werror=undef-prefix -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -emit-obj -mrelax-all --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name empty.cc -mrelocation-model pic -pic-level 2 -mframe-pointer=all -ffp-contract=on -fno-rounding-math -funwind-tables=2 -fcompatibility-qualified-id-block-type-checking -fvisibility-inlines-hidden-static-local-var -target-cpu penryn -tune-cpu generic -mllvm -treat-scalable-fixed-error-as-warning -debugger-tuning=lldb -target-linker-version 764 -v -fcoverage-compilation-dir= -resource-dir /usr/local/Cellar/llvm/14.0.6_1/lib/clang/14.0.6 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -stdlib=libc++ -internal-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/c++/v1 -internal-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/local/include -internal-isystem /usr/local/Cellar/llvm/14.0.6_1/lib/clang/14.0.6/include -internal-externc-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include -fdeprecated-macro -fdebug-compilation-dir= -ferror-limit 19 -stack-protector 1 -fblocks -fencode-extended-block-signature -fregister-global-dtors-with-atexit -fgnuc-version=4.2.1 -fcxx-exceptions -fexceptions -fmax-type-align=16 -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /var/folders/bs/zp6qt5sd2_z9yh6cy2kktbp80000gp/T/foo-9bee4e.o -x c++ empty.cc
 "/usr/local/Cellar/llvm/14.0.6_1/bin/ld64.lld" -demangle -no_deduplicate -dynamic -arch x86_64 -platform_version macos 12.0.0 12.0.0 -syslibroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -o /dev/null /var/folders/bs/zp6qt5sd2_z9yh6cy2kktbp80000gp/T/foo-9bee4e.o --start-lib --end-lib -lSystem /usr/local/Cellar/llvm/14.0.6_1/lib/clang/14.0.6/lib/darwin/libclang_rt.osx.a
```
This in turn causes a `-fuse-ld=-debugger-tuning=lldb` flag to be set, which results in the following error (since of course `-debugger-tuning=lldb` is not a valid linker path):
```
clang-14: error: invalid linker name in argument '-fuse-ld=-debugger-tuning=lldb'
```
I think the simplest way to fix this would be to check the flag ends with `.lld`, rather than check if the flag contains `lld`.

### Which operating system are you running Bazel on?

Mac

### What is the output of `bazel info release`?

Starting local Bazel server and connecting to it... (15:41:06) WARNING: info command does not support starlark options. Ignoring options: [--@io_bazel_rules_go//go/config:linkmode=pie] release 5.2.0

**Comment by zakcutner** [comment]
Hello @sgowroji! This could be reproduced on any Bazel project that builds some C/C++ code, on a system with `-debugger-tuning=lldb` set (I'm not sure what causes this to be set by default on my system). Let me know if there are any more details that would be helpful.

**Comment by keith** [comment]
can you test this branch? https://github.com/bazelbuild/bazel/pull/15006 it should be fixed here

**Comment by zakcutner** [comment]
Thanks @keith! I just tested your branch and it works great 😄

**Comment by keith** [comment]
great! hopefully it will be reviewed at some point

**Comment by dom96** [comment]
So it seems I have found a workaround: installing Xcode. On a macOS without Xcode I kept getting this problem (even when compiling the sample C++ hello world bazel example), once I installed Xcode and launched it (this part is important to finish installation) the sample C++ hello world example compiles fine.

I guess bazel completely side-steps this function when Xcode is installed. Though the bug is still in there and should be resolved IMO.
